spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -193,28 +194,32 @@ public void destroy() throws Exception { //NOSONAR
 
 	@Override
 	public void start() {
-		this.running = true;
+		this.running.compareAndSet(false, true);
```

Comment:
I believe just `this.running.set(true);` is fully enough here. Looks like logic in this method is fully idempotent. Well, just because we do nothing more 😄